multidimensional-pod-autoscaler/KEP.md

```diff
@@ -0,0 +1,499 @@
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+<!-- - <test>: <link to test coverage> -->
+
+Integration tests are to be added in the beta version.
+
+#### End-to-End Tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+<!-- - <test>: <link to test coverage> -->
+
+End-to-end tests are to be added in the beta version.
+
+## Production Readiness Review Questionnaire
+
+<!--
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+#### How can this feature be enabled / disabled in a live cluster?
+
+MPA can be enabled by checking the prerequisite and executing `./deploy/mpa-up.sh`.
+
+#### Does enabling the feature change any default behavior?
+
+No.
+
+#### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+MPA can be disabled by executing `./deploy/mpa-down.sh`.
+
+#### What happens if we reenable the feature if it was previously rolled back?
+
+No impact will happen because everytime MPA is enabled it is a full new reset and restart of MPA.
+
+#### Are there any tests for feature enablement/disablement?
+
+End-to-end test of MPA will be included in the beta version.
+
+### Dependencies
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+#### Does this feature depend on any specific services running in the cluster?
+
+MPA relies on cluster-level service metrics-server and the API server needs to support Mutating Webhooks.
+
+### Scalability
```

Comment:
We call out some concerns around scalability/frequency of updates and potential impacts on etcd higher up, could we make a first pass at addressing some of the scalability questions based on the thinking you've already done here?